xmppserver/src/main/java/org/jivesoftware/openfire/http/HttpBindManager.java

```diff
@@ -757,12 +766,12 @@ protected ServletContextHandler createStaticContentHandler()
     }
 
     /**
-     * Adds a Jetty handler to be added to the embedded web server that is used to expose BOSH (HTTP-bind)
-     * functionality.
+     * Adds a Jetty handler to be added to the embedded web server that is used to expose Openfire's public
+     * web-bindings (eg: BOSH / HTTP-bind and websocket).
      *
      * @param handler The handler (cannot be null).
      */
-    public void addJettyHandler( Handler handler )
+    public void addJettyHandler(final ContextHandler handler)
```

Comment:
Either leave this as a `org.eclipse.jetty.server.Handler` or provide a new `addJettyHandler(org.eclipse.jetty.server.Handler handler)` as a secondary implementation allowing your users to enable other jetty features (like GzipHandler, CompressionHandler, QoSHandler, RewriteHandler, etc)

Reply:
It there any reason you don't want your users to just add Servlets or Filters to your existing context instead?

Reply:
This has historically been done to somewhat isolate the functionality that is being provided by individual plugins, both from each-other, as well as from the functionality that's provided by Openfire itself on that web context.

Reply:
Do you prevent two plugins from using the same overlapping context selection rules?
A ContextHandler is selected from the ContextHandlerCollection based on ...

* The context-path declared on the ContextHandler
* The incoming path-in-context for request when it reaches the ContextHandlerCollection
* The virtual-host declarations (either direct, or named)

To oversimplify this ...
If more than one ContextHandler entry results from the selection list above, then the resulting list is sorted, and then each entry is given the request to process following Jetty Core Handler rules, stopping when one of the handlers declares itself to be the one handling the generation of the response.

Is that something you attempt to prevent? or let the plugins do what they want?